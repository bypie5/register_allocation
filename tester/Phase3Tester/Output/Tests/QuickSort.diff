--- SelfTestCases/QuickSort.out	2020-03-05 22:31:29.000000000 -0800
+++ ./Output/Tests/QuickSort.out	2020-03-09 00:14:29.000000000 -0700
@@ -1,22 +1,5 @@
-20
-7
-12
-18
-2
-11
-6
-9
-19
-5
-9999
-2
-5
-6
-7
-9
-11
-12
-18
-19
-20
-0
+--- Error (ProgramFault) ---
+require function address, found integer10
+  Main() line 20
+  QS.Start() line 39
+  QS.Init() line 341
