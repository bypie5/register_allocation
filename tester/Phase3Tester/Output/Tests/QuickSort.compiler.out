func Main [in 0, out 0, local 3]
local[0] = $s0
local[1] = $s1
local[2] = $s2
$s0 = HeapAllocZ(12)
[$s0+0] = :vmt_QS
if $s0 goto :null1
Error("null pointer")
null1:
$s1 = [$s0+0]
$s1 = [$s1+0]
$a0 = $s0
$a1 = 10
call $s1
$s2 = $v0
PrintIntS($s2)
$s0 = local[0]
$s1 = local[1]
$s2 = local[2]
ret

func QS.Start [in 0, out 0, local 4]
local[0] = $s0
local[1] = $s1
local[2] = $s2
local[3] = $s3
$s1 = $a0
$s0 = $a1
$s2 = [$s1+0]
$s2 = [$s2+12]
$a0 = $s1
$a1 = $s0
call $s2
$s3 = $v0
$s0 = [$s1+0]
$s0 = [$s0+8]
$a0 = $s1
call $s0
$s2 = $v0
PrintIntS(9999)
$s0 = [$s1+8]
$s2 = Sub($s0 1)
$s0 = [$s1+0]
$s0 = [$s0+4]
$a0 = $s1
$a1 = 0
$a2 = $s2
call $s0
$s2 = $v0
$s0 = [$s1+0]
$s0 = [$s0+8]
$a0 = $s1
call $s0
$s2 = $v0
$s0 = local[0]
$s1 = local[1]
$s2 = local[2]
$s3 = local[3]
ret

func QS.Sort [in 0, out 0, local 7]
local[0] = $s0
local[1] = $s1
local[2] = $s2
local[3] = $s3
local[4] = $s4
local[5] = $s5
local[6] = $s6
$s2 = $a0
$s0 = $a1
$s1 = $a2
$s3 = 0
$s3 = LtS($s0 $s1)
if $s3 goto :if1_else
$s0 = [$s2+4]
if $s0 goto :null2
Error("null pointer")
null2:
$s2 = [$s0+0]
$s2 = Lt($s1 $s2)
if $s2 goto :bounds1
Error("array index out of bounds")
bounds1:
$s1 = MulS($s3 4)
$s1 = Add($s1 $s0)
$s0 = [$s1+4]
$s0 = Sub($s2 1)
$s0 = $s3
$s0 = 1
while1_top:
if $s0 goto :while1_end
$s0 = 1
while2_top:
if $s0 goto :while2_end
$s1 = Add($s1 1)
$s0 = [$s2+4]
if $s0 goto :null3
Error("null pointer")
null3:
$s2 = [$s0+0]
$s2 = Lt($s1 $s2)
if $s2 goto :bounds2
Error("array index out of bounds")
bounds2:
$s2 = MulS($s0 4)
$s2 = Add($s2 $s1)
$s0 = [$s2+4]
$s1 = LtS($s0 $s3)
$s0 = Sub(1 $s1)
if $s0 goto :if2_else
$s0 = 0
goto :if2_end
if2_else:
$s0 = 1
if2_end:
goto :while2_top
while2_end:
$s0 = 1
while3_top:
if $s0 goto :while3_end
$s1 = Sub($s1 1)
$s0 = [$s2+4]
if $s0 goto :null4
Error("null pointer")
null4:
$s2 = [$s0+0]
$s2 = Lt($s1 $s2)
if $s2 goto :bounds3
Error("array index out of bounds")
bounds3:
$s2 = MulS($s0 4)
$s2 = Add($s2 $s1)
$s0 = [$s2+4]
$s1 = LtS($s3 $s0)
$s0 = Sub(1 $s1)
if $s0 goto :if3_else
$s0 = 0
goto :if3_end
if3_else:
$s0 = 1
if3_end:
goto :while3_top
while3_end:
$s1 = [$s0+4]
if $s1 goto :null5
Error("null pointer")
null5:
$s2 = [$s0+0]
$s2 = Lt($s1 $s2)
if $s2 goto :bounds4
Error("array index out of bounds")
bounds4:
$s2 = MulS($s0 4)
$s2 = Add($s2 $s1)
$s0 = [$s2+4]
$s0 = [$s3+4]
if $s0 goto :null6
Error("null pointer")
null6:
$s2 = [$s0+0]
$s2 = Lt($s1 $s2)
if $s2 goto :bounds5
Error("array index out of bounds")
bounds5:
$s2 = MulS($s0 4)
$s2 = Add($s2 $s1)
$s0 = [$s3+4]
if $s0 goto :null7
Error("null pointer")
null7:
$s2 = [$s0+0]
$s2 = Lt($s1 $s2)
if $s2 goto :bounds6
Error("array index out of bounds")
bounds6:
$s2 = MulS($s0 4)
$s2 = Add($s2 $s1)
$s0 = [$s2+4]
[$s3+4] = $s0
$s0 = [$s4+4]
if $s0 goto :null8
Error("null pointer")
null8:
$s2 = [$s0+0]
$s2 = Lt($s1 $s2)
if $s2 goto :bounds7
Error("array index out of bounds")
bounds7:
$s3 = MulS($s4 4)
$s3 = Add($s3 $s0)
[$s3+4] = $s1
$s0 = Add($s2 1)
$s1 = LtS($s4 $s0)
if $s1 goto :if4_else
$s0 = 0
goto :if4_end
if4_else:
$s0 = 1
if4_end:
goto :while1_top
while1_end:
$s1 = [$s0+4]
if $s1 goto :null9
Error("null pointer")
null9:
$s2 = [$s0+0]
$s2 = Lt($s1 $s2)
if $s2 goto :bounds8
Error("array index out of bounds")
bounds8:
$s2 = MulS($s0 4)
$s2 = Add($s2 $s1)
$s0 = [$s3+4]
if $s0 goto :null10
Error("null pointer")
null10:
$s2 = [$s0+0]
$s2 = Lt($s1 $s2)
if $s2 goto :bounds9
Error("array index out of bounds")
bounds9:
$s2 = MulS($s0 4)
$s2 = Add($s2 $s1)
$s0 = [$s2+4]
[$s3+4] = $s0
$s0 = [$s4+4]
if $s0 goto :null11
Error("null pointer")
null11:
$s2 = [$s0+0]
$s2 = Lt($s1 $s2)
if $s2 goto :bounds10
Error("array index out of bounds")
bounds10:
$s2 = MulS($s0 4)
$s2 = Add($s2 $s1)
$s0 = [$s3+4]
if $s0 goto :null12
Error("null pointer")
null12:
$s2 = [$s0+0]
$s2 = Lt($s1 $s2)
if $s2 goto :bounds11
Error("array index out of bounds")
bounds11:
$s2 = MulS($s0 4)
$s2 = Add($s2 $s1)
$s0 = [$s2+4]
[$s3+4] = $s0
$s0 = [$s4+4]
if $s0 goto :null13
Error("null pointer")
null13:
$s2 = [$s0+0]
$s2 = Lt($s1 $s2)
if $s2 goto :bounds12
Error("array index out of bounds")
bounds12:
$s2 = MulS($s5 4)
$s2 = Add($s2 $s0)
[$s2+4] = $s1
$s0 = [$s6+0]
$s0 = [$s0+4]
$s1 = Sub($s4 1)
$a0 = $s6
$a1 = $s3
$a2 = $s1
call $s0
$s2 = $v0
$s0 = [$s6+0]
$s0 = [$s0+4]
$s1 = Add($s4 1)
$a0 = $s6
$a1 = $s1
$a2 = $s5
call $s0
$s2 = $v0
goto :if1_end
if1_else:
$s0 = 0
if1_end:
$s0 = local[0]
$s1 = local[1]
$s2 = local[2]
$s3 = local[3]
$s4 = local[4]
$s5 = local[5]
$s6 = local[6]
ret

func QS.Print [in 0, out 0, local 4]
local[0] = $s0
local[1] = $s1
local[2] = $s2
local[3] = $s3
$s0 = $a0
$s0 = 0
while4_top:
$s1 = [$s2+8]
$s3 = LtS($s0 $s1)
if $s3 goto :while4_end
$s0 = [$s2+4]
if $s0 goto :null14
Error("null pointer")
null14:
$s2 = [$s0+0]
$s2 = Lt($s1 $s2)
if $s2 goto :bounds13
Error("array index out of bounds")
bounds13:
$s1 = MulS($s2 4)
$s1 = Add($s1 $s0)
$s0 = [$s1+4]
PrintIntS($s0)
$s2 = Add($s2 1)
goto :while4_top
while4_end:
$s0 = local[0]
$s1 = local[1]
$s2 = local[2]
$s3 = local[3]
ret

func QS.Init [in 0, out 0, local 4]
local[0] = $s0
local[1] = $s1
local[2] = $s2
local[3] = $s3
$s2 = $a0
$s1 = $a1
[$s2+8] = $s1
$a0 = $s1
call $s0
$s3 = $v0
[$s2+4] = $s3
$s0 = [$s2+4]
if $s0 goto :null15
Error("null pointer")
null15:
$s1 = [$s0+0]
$s1 = Lt(0 $s1)
if $s1 goto :bounds14
Error("array index out of bounds")
bounds14:
$s1 = MulS(0 4)
$s1 = Add($s1 $s0)
[$s1+4] = 20
$s0 = [$s2+4]
if $s0 goto :null16
Error("null pointer")
null16:
$s1 = [$s0+0]
$s1 = Lt(1 $s1)
if $s1 goto :bounds15
Error("array index out of bounds")
bounds15:
$s1 = MulS(1 4)
$s1 = Add($s1 $s0)
[$s1+4] = 7
$s0 = [$s2+4]
if $s0 goto :null17
Error("null pointer")
null17:
$s1 = [$s0+0]
$s1 = Lt(2 $s1)
if $s1 goto :bounds16
Error("array index out of bounds")
bounds16:
$s1 = MulS(2 4)
$s1 = Add($s1 $s0)
[$s1+4] = 12
$s0 = [$s2+4]
if $s0 goto :null18
Error("null pointer")
null18:
$s1 = [$s0+0]
$s1 = Lt(3 $s1)
if $s1 goto :bounds17
Error("array index out of bounds")
bounds17:
$s1 = MulS(3 4)
$s1 = Add($s1 $s0)
[$s1+4] = 18
$s0 = [$s2+4]
if $s0 goto :null19
Error("null pointer")
null19:
$s1 = [$s0+0]
$s1 = Lt(4 $s1)
if $s1 goto :bounds18
Error("array index out of bounds")
bounds18:
$s1 = MulS(4 4)
$s1 = Add($s1 $s0)
[$s1+4] = 2
$s0 = [$s2+4]
if $s0 goto :null20
Error("null pointer")
null20:
$s1 = [$s0+0]
$s1 = Lt(5 $s1)
if $s1 goto :bounds19
Error("array index out of bounds")
bounds19:
$s1 = MulS(5 4)
$s1 = Add($s1 $s0)
[$s1+4] = 11
$s0 = [$s2+4]
if $s0 goto :null21
Error("null pointer")
null21:
$s1 = [$s0+0]
$s1 = Lt(6 $s1)
if $s1 goto :bounds20
Error("array index out of bounds")
bounds20:
$s1 = MulS(6 4)
$s1 = Add($s1 $s0)
[$s1+4] = 6
$s0 = [$s2+4]
if $s0 goto :null22
Error("null pointer")
null22:
$s1 = [$s0+0]
$s1 = Lt(7 $s1)
if $s1 goto :bounds21
Error("array index out of bounds")
bounds21:
$s1 = MulS(7 4)
$s1 = Add($s1 $s0)
[$s1+4] = 9
$s0 = [$s2+4]
if $s0 goto :null23
Error("null pointer")
null23:
$s1 = [$s0+0]
$s1 = Lt(8 $s1)
if $s1 goto :bounds22
Error("array index out of bounds")
bounds22:
$s1 = MulS(8 4)
$s1 = Add($s1 $s0)
[$s1+4] = 19
$s0 = [$s2+4]
if $s0 goto :null24
Error("null pointer")
null24:
$s1 = [$s0+0]
$s1 = Lt(9 $s1)
if $s1 goto :bounds23
Error("array index out of bounds")
bounds23:
$s1 = MulS(9 4)
$s1 = Add($s1 $s0)
[$s1+4] = 5
$s0 = local[0]
$s1 = local[1]
$s2 = local[2]
$s3 = local[3]
ret

func AllocArray [in 0, out 0, local 3]
local[0] = $s0
local[1] = $s1
local[2] = $s2
$s0 = $a0
$s1 = MulS($s0 4)
$s1 = Add($s1 4)
$s2 = HeapAllocZ($s1)
[$s2+0] = $s0
$v0 = $s2
$s0 = local[0]
$s1 = local[1]
$s2 = local[2]
ret

